### PR TITLE
feat: achievement modal

### PR DESCRIPTION
**Pull request by LucasFalkowsky**
closes https://github.com/corona-school/project-user/issues/978

Modal Components for Desktop and Mobile view were implemented.
Stories for Modal and Desktop Components were added.

### CODE REVIEW

File: src/components/achievements/achievementCard/achievementCard.stories.mdx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Canvas, Meta, Story } from '@storybook/addon-docs';
-import AchievementCard, { CardState, ActionTypes } from './achievementCard';
+import AchievementCard from './achievementCard';
```
Comment (LucasFalkowsky) [inline src/components/achievements/achievementCard/achievementCard.stories.mdx:2]: This renaming was done, due to the changed purpose of this enum. It is used for everything achievement related now.

File: src/components/achievements/cosmetics/newAchievementShine.tsx [hunk 2]
```diff
@@ -38,9 +39,19 @@ const NewAchievementShine: React.FC<NewAchievementShineProps> = ({ isMobile }) =
                 positionTop: animatedShineProps.props.positionTop / 2,
             };
         }
+        if (isLarge) {
```
Comment (LucasFalkowsky) [inline src/components/achievements/cosmetics/newAchievementShine.tsx:42]: if isLarge is passed as a prop we are scaling up the animation

File: src/components/achievements/modals/achievementModal.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                            <Box>
+                                <Box marginLeft={'64px'} width={'142px'}>
+                                    <PolaroidImageContainer
+                                        image={achievementState === AchievementState.COMPLETED ? image : undefined}
+                                        alternativeText={alternativeText || ''}
+                                        isLarge
+                                    />
+                                </Box>
+                                {newAchievement && (
+                                    <Box position={'absolute'} top={'-40px'} left={'-40px'}>
```
Comment (LucasFalkowsky) [inline src/components/achievements/modals/achievementModal.tsx:49]: due to a new scaling the position of this Animation was adjusted

File: src/components/achievements/modals/achievementModal.tsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    image,
+    alternativeText,
+    achievementState,
+    achievementType,
+}) => {
+    const { t } = useTranslation();
+    const [showModal, setShowModal] = useState(true);
+
+    const isMobile = useBreakpointValue({ base: true, sm: true, md: false });
+    const isTablet = useBreakpointValue({ md: true, lg: false });
```
Comment (LomyW) [inline src/components/achievements/modals/achievementModal.tsx:56]: also here, no tablet breakpoint for consistency

File: src/components/achievements/modals/achievementModal.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            isOpen={showModal}
+            onClose={() => setShowModal(false)}
+            width="100vw"
+            height="100vh"
+            justifyContent={isMobile ? 'normal' : 'center'}
+            alignItems={isMobile ? 'normal' : 'center'}
+            display={isMobile ? 'flex' : 'grid'}
+        >
+            <Modal.Body
+                width={isMobile || isTablet ? '100%' : '820px'}
```
Comment (LomyW) [inline src/components/achievements/modals/achievementModal.tsx:72]: also for better readability and consitency, we should create separate breakpoint values for the different props, like in the component `AppointmentCard.tsx`

File: src/components/achievements/modals/achievementModal.tsx [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+                            </VStack>
+                            {(isMobile || isTablet) && (
+                                <VStack width="100%" alignItems="center" space="8px">
+                                    <Box width="80%">
+                                        {achievementState === AchievementState.COMPLETED ? (
+                                            <Box width="100%" display="flex" alignItems="center">
+                                                {newAchievement ? (
+                                                    <AchievementBadge isInline />
+                                                ) : (
+                                                    <Text color={Theme.colors.primary[500]} textAlign="center">
```
Comment (LomyW) [inline src/components/achievements/modals/achievementModal.tsx:161]: you could also user `color={"primary.500"}` here

File: src/components/achievements/progressIndicators/indicatorStep.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    achievementState?: AchievementState;
+    isActive?: boolean;
+    isInactive?: boolean;
+};
+
+const IndicatorStep: React.FC<IndicatorStepProps> = ({ step, maxSteps, description, achievementState, isActive, isInactive }) => {
+    const offsetPerStep = 100 / (maxSteps - 1);
+    const offset = offsetPerStep * step;
+    const textColor = achievementState === AchievementState.COMPLETED ? Theme.colors.white : Theme.colors.primary[900];
+    console.log(step, maxSteps, description, achievementState, isActive, isInactive);
```
Comment (LomyW) [inline src/components/achievements/progressIndicators/indicatorStep.tsx:19]: remove log

File: src/components/achievements/progressIndicators/indicatorStep.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            <VStack
+                alignItems="center"
+                justifyContent="center"
+                width="20px"
+                height="20px"
+                borderRadius="50%"
+                backgroundColor={achievementState !== AchievementState.COMPLETED && isInactive ? 'gray.100' : 'primary.500'}
+                outlineStyle="solid"
+                outlineColor="rgba(0, 169, 145, 0.2)"
+                outlineWidth={isInactive ? '0' : '3px'}
```
Comment (LomyW) [inline src/components/achievements/progressIndicators/indicatorStep.tsx:31]: does `outlineWidth` prop exist for VStacks? Get an error there

File: src/components/achievements/progressIndicators/indicatorStep.tsx (hunk 7 above)
Comment (LomyW) [inline src/components/achievements/progressIndicators/indicatorStep.tsx:19]: also here use `"primary.900"`, maybe direct in the text prop

File: src/components/achievements/progressIndicators/indicatorStep.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+                {achievementState !== AchievementState.COMPLETED && (isActive || isInactive) ? (
+                    <Text fontSize={'xs'} color={isInactive ? Theme.colors.gray[500] : Theme.colors.white}>
+                        {step + 1}
+                    </Text>
+                ) : (
+                    <Check />
+                )}
+            </VStack>
+            <Text
+                position={'absolute'}
```
Comment (LomyW) [inline src/components/achievements/progressIndicators/indicatorStep.tsx:42]: also here for props you dont need a extra object `{}`

File: src/components/achievements/streak/StreakImage.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    alternativeText: string;
+    size: StreakImageSize;
+};
+
+const StreakImage: React.FC<StreakImageProps> = ({ streak, image, alternativeText, size }) => {
+    const { t } = useTranslation();
+    return (
+        <VStack justifyContent="center" alignItems="center" width={size} height={size}>
+            <VStack position="absolute" width={getStreakImageSize(size)} height="100%">
+                {/* <img width={'100%'} src={image} alt='' aria-hidden /> */}
```
Comment (LomyW) [inline src/components/achievements/streak/StreakImage.tsx:20]: why is this commented out?

**Review by LomyW — CHANGES_REQUESTED** [review]
All in all, beautiful components. We should just make sure that we maintain consistency and leave out the breakpoint for Tablet.

**Review by LomyW — CHANGES_REQUESTED** [review]
All in all, beautiful components. We should just make sure that we maintain consistency and leave out the breakpoint for Tablet.

File: src/components/achievements/helpers/achievement-image-helper.ts [hunk 11]
```diff
@@ -0,0 +1,87 @@
+import { PolaroidImageSize, PuzzleImageSize, ShineOffset, ShineSize } from '../types';
```
[the remark this answers is not in the document]
Comment (LucasFalkowsky) [reply src/components/achievements/helpers/achievement-image-helper.ts:1]: it should be in lower case ofc. Made that mistake for sure. Ill refactor this when we merged.

File: src/components/achievements/achievementCard/achievementCard.tsx [hunk 12]
```diff
@@ -58,41 +49,71 @@ const AchievementCard: React.FC<AchievementCardProps> = ({
...
+                }
             >
-                {newAchievement && cardState === CardState.COMPLETED && (
+                {newAchievement && achievementState === AchievementState.COMPLETED && (
                     <>
                         <AchievementBadge isMobile={isMobile} />
-                        <NewAchievementShine isMobile={isMobile} />
+                        <VStack position="absolute" zIndex={1} justifyContent="center" alignItems="center">
+                            <VStack
+                                width={isMobile ? '64px' : getPolaroidImageSize(isMobile, isMobile, true)}
```
[the remark this answers is not in the document]
Comment (LucasFalkowsky) [reply src/components/achievements/achievementCard/achievementCard.tsx:93]: the Component NewAchievementShine treats these states the same.

File: src/components/achievements/progressIndicators/indicatorStep.tsx (hunk 8 above)
Comment (LucasFalkowsky) [reply src/components/achievements/progressIndicators/indicatorStep.tsx:31]: it did work, even though it was underlined. But I fixed this by adding a border to the parent element.

File: src/components/achievements/streak/StreakImage.tsx (hunk 10 above)
Comment (LucasFalkowsky) [reply src/components/achievements/streak/StreakImage.tsx:20]: for when it is decided, what to use. currently we have a dummy image in this component.

File: src/components/achievements/polaroid/completePolaroid.tsx [hunk 13]
```diff
@@ -1,29 +1,27 @@
...
             justifyContent="flex-start"
             alignItems="center"
             shadow="0px 2px 4px 0px rgba(0, 0, 0, 0.25)"
             backgroundColor="white"
-            width={isMobile ? '60px' : '136px'}
-            height={isMobile ? '80px' : '184px'}
-            borderRadius={isMobile ? '2px' : '4px'}
-            paddingTop={isMobile ? '4px' : '7.5px'}
+            width={size}
+            height={`calc(${size} * 1.35)`}
```
Comment (LomyW) [inline src/components/achievements/polaroid/completePolaroid.tsx:19]: for what is the calc and why do wie calc inline?

File: src/components/achievements/polaroid/completePolaroid.tsx [hunk 14]
```diff
@@ -1,29 +1,27 @@
...
+            borderRadius={getPolaroidBorderRadius(size)}
+            paddingTop={`calc(${getPolaroidBorderRadius(size)} * 2)`}
+            top={size === PolaroidImageSize.SMALL ? 0 : getPolaroidOffset(size)}
```
Comment (LomyW) [inline src/components/achievements/polaroid/completePolaroid.tsx:22]: why ternary? `getPolaroidOffset()` also checks small image size or?

File: src/components/achievements/modals/achievementModal.tsx [hunk 15]
```diff
@@ -0,0 +1,247 @@
...
+                                    fontSize={isMobile || isTablet ? 'xl' : '4xl'}
+                                    fontWeight="bold"
+                                    lineHeight={isMobile || isTablet ? '16px' : '36px'}
+                                    textAlign={isMobile || isTablet ? 'center' : 'left'}
+                                    color={textColor}
+                                >
+                                    {name}
+                                </Text>
+                                {!isMobile && !isTablet && (
+                                    <Text color={textColor} numberOfLines={7}>
```
Comment (LomyW) [inline src/components/achievements/modals/achievementModal.tsx:147]: soo we dont need `ellipsizeMode` here?

File: src/components/achievements/polaroid/completePolaroid.tsx (hunk 13 above)
Comment (LucasFalkowsky) [reply src/components/achievements/polaroid/completePolaroid.tsx:19]: this calc is taking the width of the polaroid (the size value) and it gives the height a 1.35 aspect ratio to this width. This ensures, the polaroid always has the right aspect ratio, no matter the scaling.

File: src/components/achievements/polaroid/completePolaroid.tsx (hunk 14 above)
Comment (LucasFalkowsky) [reply src/components/achievements/polaroid/completePolaroid.tsx:22]: true. I could just take the polaroidOffset return value. ' _ '*

**Review by LomyW — COMMENTED** [review]
nice work! ✨

File: src/components/achievements/modals/achievementModal.tsx (hunk 15 above)
Comment (LucasFalkowsky) [reply src/components/achievements/modals/achievementModal.tsx:147]: no. It makes the ... without the ellipsizeMode. Don't know why but it does.

**Review by LomyW — APPROVED** [review]

**Pull request merged**